src/Native/Unix/get-prerequisite.sh

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env bash
+
+# Downloads the package corresponding to the declared version of the given prerequisite name, 
+# and expands the downloaded package to Tools/Downloads/ folder in the repository root.
+
+# Exit 1 if unable to acquire the declared version of the prerequisite.
+
+# Arguments:
+#   1. Prerequisite name.
+#   2. (Optional) Repository root path. If not specified then, will be determined as 3 levels up the current working folder.
+#   3. (Optional) Declared version of the prerequisite. If not specified, declared version will be determined by invoking ./get-declared-prerequisite-version.sh.
+#   4. (Optional) URL of the prerequisite package from where the package will be downloaded.
+
+if [ -z "$1" ]; then
+    echo "Argument passed as prerequisite name is empty. Please provide a non-empty string."
+    exit 1
+else
+    prerequisiteName="$1"
+fi
+
+if [[ ! -z "$2" && -d "$2" ]]; then
+    repoRoot="$( cd "$2" && pwd )"
+fi
+
+if [ ! -z "$3" ]; then
+    declaredVersion="$3"
+fi
+
+if [ ! -z "$4" ]; then
+    downloadUrl="$4"
+fi
+
+# Get declared version.
+get_declared_version()
+{
+    if [ -z "$declaredVersion" ]; then
+        declaredVersion=$("$repoRoot/src/Native/Unix/get-declared-prerequisite-version.sh" "$prerequisiteName" "$repoRoot")
+
+        if [ $? -eq 1 ]; then
+            echo "$declaredVersion"
+            exit 1
+        fi
+    fi
+
+    echo "Declared version of $prerequisiteName is $declaredVersion"
+}
+
+# Determine the package name based on declared version and OS.
+get_package_name()
+{
+    packageName=$("$repoRoot/src/Native/Unix/get-prerequisite-package-name.sh" "$prerequisiteName" "$declaredVersion")
+
+    if [ $? -eq 1 ]; then
+        echo "$packageName"
+        exit 1
+    fi
+
+    packageNameWithExtension=$packageName".tar.gz"
+}
+
+# Prepare the URL from where the package can be downloaded.
+get_prerequisite_package_url()
+{
+    # TODO: 
+    #   Do not download directly from internet. 
+    #   Follow the practice described at https://www.1eswiki.com/wiki/Introducing_OSS_Component_Governance
+    #   Likely we will host the prerequisite at https://ossmsft.visualstudio.com/, and the below logic will change.
+
+    if [ -z "$downloadUrl" ]; then
+        # Prepare the download URL. For example, https://cmake.org/files/v3.7/cmake-3.7.2-Darwin-x86_64.tar.gz
+        # Determine the version fragment. For example v3.7 in https://cmake.org/files/v3.7/cmake-3.7.2-Darwin-x86_64.tar.gz
+        urlVersionFragment="v"$(echo $declaredVersion | cut -d '.' -f 1,2)
+        downloadUrl="https://cmake.org/files/"$urlVersionFragment"/"$packageNameWithExtension
```

Comment:
I realize this is only for cmake right now but we definitely need to make this a more general endpoint at some point.